sky/serve/service.py

```diff
@@ -236,13 +239,12 @@ def _get_host():
         serve_state.set_service_status_and_active_versions(
             service_name, serve_state.ServiceStatus.SHUTTING_DOWN)
     finally:
-        process_to_kill: List[multiprocessing.Process] = []
-        if load_balancer_process is not None:
-            process_to_kill.append(load_balancer_process)
-        if controller_process is not None:
-            process_to_kill.append(controller_process)
         # Kill load balancer process first since it will raise errors if failed
         # to connect to the controller. Then the controller process.
+        process_to_kill = [
+            proc for proc in [load_balancer_process, controller_process]
+            if proc is not None
+        ]
         subprocess_utils.kill_children_processes(
             [process.pid for process in process_to_kill], force=True)
```

Comment:
```suggestion
        pids_to_kill = [
            proc.pid for proc in [load_balancer_process, controller_process]
            if proc is not None
        ]
        subprocess_utils.kill_children_processes(
            pids_to_kill, force=True)
```
nit: lets merge the two list comprehension together ;)

Reply:
Thanks for the suggestion! However, we still need the `process_to_kill` list for the `process.join()` call in the following lines, so I can't merge the list comprehensions.

Reply:
Good point!